File: Problem/LeetCode/C++/Maximum_Number_of_Balloons.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int maxNumberOfBalloons(string text) {
        map <char, int> arr;
        for (int i = 0; i < text.length(); i++) {
            arr[text[i]] += 1;
        }
        string str = "balloon";
        int res = 1000000;
        for (char c : str){
            if (c == 'l' || c == 'o'){
                res = min(res, arr[c] / 2);
            }
            else res = min(res, arr[c]);
        }
        return res;
    }
};

int main(){
    Solution test;
    cout << test.maxNumberOfBalloons("loonbalxballpoon");

    return 0;
}

File: Problem/LeetCode/Python/Maximum Number of Balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        from collections import defaultdict
        arr = defaultdict(int)
        # n = len(text)

        for c in text:
             arr[c] += 1;

        res = 1000000
        for c in ['b', 'a', 'n']:
            res = min(res, arr[c])
        for c in ['l', 'o']:
            res = min(res, arr[c] // 2)
        
        # print(arr)

        return res

test = Solution()
print(test.maxNumberOfBalloons("nlaebolko"))